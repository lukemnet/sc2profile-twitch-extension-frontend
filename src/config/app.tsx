const app = {
  defaultLanguage: process.env.REACT_APP_DEFAULT_LANGUAGE || 'en',
  refreshInterval: Number(process.env.REACT_APP_UPDATE_INTERVAL) || 480000,
  enableTwitchAuth: (process.env.REACT_APP_ENABLE_TWITCH_AUTH === 'true') || false,
  maxProfiles: Number(process.env.REACT_APP_MAXIMUM_PLAYER_PROFILE_COUNT) || 3,
};

export default app;
