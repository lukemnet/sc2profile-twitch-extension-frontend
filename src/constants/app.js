const API_URL = process.env.REACT_APP_SC2PTE_API_URL;

module.exports = {
  api: {
    url: API_URL,
    version: 'v1.1',
  },
  config: {
    fetch: {
      endpoint: 'config/get',
      method: 'GET',
    },
    save: {
      endpoint: 'config/save',
      method: 'POST',
    },
  },
  viewer: {
    fetch: {
      endpoint: 'viewer/get',
      method: 'GET',
    },
  },
};
