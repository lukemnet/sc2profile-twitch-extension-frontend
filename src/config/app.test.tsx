import app from './app';

it('exports app config object', () => {
  expect(app).toMatchSnapshot();
});
