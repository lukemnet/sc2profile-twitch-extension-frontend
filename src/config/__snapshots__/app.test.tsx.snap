// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`exports app config object 1`] = `
Object {
  "defaultLanguage": "en",
  "enableTwitchAuth": false,
  "maxProfiles": 3,
  "refreshInterval": 480000,
}
`;
